xsub.go

```diff
@@ -7,18 +7,24 @@ package zmq4
 import (
 	"context"
 	"net"
+	"strings"
+	"sync"
 )
 
 // NewXSub returns a new XSUB ZeroMQ socket.
 // The returned socket value is initially unbound.
 func NewXSub(ctx context.Context, opts ...Option) Socket {
-	xsub := &xsubSocket{newSocket(ctx, XSub, opts...)}
+	xsub := &xsubSocket{newSocket(ctx, XSub, opts...), sync.RWMutex{}, make(map[string]struct{})}
+	xsub.sck.r = newQReader(xsub.sck.ctx)
+	xsub.topics = make(map[string]struct{})
 	return xsub
 }
 
 // xsubSocket is a XSUB ZeroMQ socket.
 type xsubSocket struct {
 	sck *socket
+	mu     sync.RWMutex
```

Comment:
please insert a blank line above `mu` to clearly delimitate the block that mutex is supposed to guard.

Reply:
Committed in issue-108.